src/components/Topology/Topology.js

```diff
@@ -0,0 +1,65 @@
+import React, { useRef, useEffect } from "react";
+import * as d3 from "d3";
+
+import { generateIconPath } from "app/utils";
+
+const isSubordinate = app => app.subordinateTo.length > 0;
+
+export default ({ modelData, width, height }) => {
+  const ref = useRef();
+
+  const applications =
+    (modelData &&
+      Object.keys(modelData.applications).map(appName => ({
+        ...modelData.applications[appName],
+        name: appName
+      }))) ||
+    [];
+
+  useEffect(() => {
+    const zoom = d3.zoom();
+
+    const topo = d3
+      .select(ref.current)
+      .attr("width", width)
+      .attr("height", height)
+      .append("g")
+      .call(zoom)
+      // 2 lines below copied....why do they work?
```

Comment:
magnets, how do they work?

Reply:
https://developer.mozilla.org/en-US/docs/Web/CSS/transform

Reply:
It was a comment to remind me to investigate how `zoom.transform` and `d3.zoomIdentity` work :)

The second line is also a transform attribute on the svg itself not a css transform.